src/include/denc.h

```diff
@@ -713,17 +713,28 @@ struct denc_traits<std::basic_string<char,std::char_traits<char>,A>> {
   {
     uint32_t len;
     denc(len, p);
-    s.clear();
-    p.copy(len, s);
+    decode_nohead(len, s, p);
   }
   template<class It>
-  static std::enable_if_t<is_const_iterator_v<It>>
-  decode_nohead(size_t len, value_type& s, It& p) {
+  static void decode_nohead(size_t len, value_type& s, It& p) {
     s.clear();
     if (len) {
       s.append(p.get_pos_add(len), len);
     }
   }
+  static void decode_nohead(size_t len, value_type& s,
+                            buffer::list::const_iterator& p) {
+    s.clear();
+    if (len) {
+      if constexpr (std::is_same_v<value_type, std::string>) {
+        s.clear();
+        p.copy(len, s);
+      } else {
+        s.append(len, 0);
+        p.copy(len, s.data());
```

Comment:
the std::string case duplicates the call to clear, and this case could replace clear+append with resize

Reply:
@cbodley thanks! fixed and repushed.